File: VTI_Galaxy_BE/src/main/java/demo/services/interfaceClass/MovieService.java
package demo.services.interfaceClass;

import demo.modal.dto.MovieDto;
import demo.modal.request.MovieRequest;

import java.util.List;

public interface MovieService {
    // get
    List<MovieDto> getAllMovies();
    MovieDto getMovieById(int id);

    // post
    MovieDto addMovie(MovieRequest request);

    // put
    MovieDto updateMovie(MovieRequest request, int id);

    // closed
    MovieDto updateStatusMovie(int id);

    // set delete
    MovieDto deleteMovie(int id);
}

File: VTI_Glaxy_FE/src/services/movieService.js
import axiosClient from "./axiosClient";

const movieService = {
  fetchMovies() {
    return axiosClient.get("/getMovies");
  },
  fetchMovieById(movieId) {
    return axiosClient.get(`/getMovieById?movieId=${movieId}`);
  },
  createMovie(movieRequest) {
    return axiosClient.post("/postMovie", movieRequest);
  },
  updateMovie(movieId, movieRequest) {
    return axiosClient.put(`/putMovie?movieId=${movieId}`, movieRequest);
  },
  deleteMovie(movieId) {
    return axiosClient.put(`/deleteMovie?movieId=${movieId}`);
  },
};

export default movieService;